code/game/objects/structures/fence.dm

```diff
@@ -89,15 +89,17 @@
 	. = TRUE
 	if(shock(user, 100))
 		return
+	if(invulnerable)
+		to_chat(user, "<span class='warning'>This fence is too strong to cut through!</span>")
+		return
 	if(!cuttable)
+		visible_message("<span class='danger'>[user] starts dismantling [src] with [W].</span>")
+		to_chat(user, "<span class='danger'>You start dismantling [src] with [W].</span>")
```

Comment:
Could combine this into one `visible_message` send from the user with a self message.
I feel like `danger` is a bit overkill for what you're doing. `warning` is more suited I'd say.
I personally see `danger` as something that might or has hurt somebody other than the user directly. `warning` is something that might potentially be bad but also has a chance to be okay and just needs a bit more attention.